### ISSUE TITLE
filter_kubernetes: [warn] could not get meta for POD fluent-bit

### ISSUE THREAD

**Issue by valentin2105** [fluent/fluent-bit#850]
## Bug Report

**Describe the bug**
Fluent-bit doesn't launch well on my IPv6-only Kubernetes cluster

**To Reproduce**
Create k8s objects from your Kubernetes repo following the doc (serviceAccount,role,binding,config,ds...) and then inspect the fluent-bit pod's logs.

**Expected behavior**
Get my logs shipped to ES. 

**Your Environment**

* Version used: fluent/fluent-bit:0.14.3
* Configuration:  using different k8s Endpoint that 'svc.default.cluster.local..'  (that work with curl) 
* Environment name and version: Kubernetes 1.10.6, IPv6-only , Calico 
* Server type and version: Debian 9.x up-to-date

Logs : 

```
[2018/10/15 03:48:57] [ info] [engine] started (pid=1)
[2018/10/15 03:48:57] [ info] [filter_kube] https=1 host=k8s-api.domain.tld port=443
[2018/10/15 03:48:57] [ info] [filter_kube] local POD info OK
[2018/10/15 03:48:57] [ info] [filter_kube] testing connectivity with API server...
[2018/10/15 03:48:57] [error] [filter_kube] upstream connection error
[2018/10/15 03:48:57] [ warn] [filter_kube] could not get meta for POD fluent-bit-t2l45
[2018/10/15 03:48:57] [ info] [http_server] listen iface=0.0.0.0 tcp_port=2020

<nothing after that>
```

**Comment by edsiper** [comment]
For some reason, Fluent Bit cannot reach Kubernetes API server. If you ssh into Fluent Bit pod are you able to connect to https://k8s-api.domain.tld ?

**Comment by hhoover** [comment]
same issue here. Fluent-bit can't handle the IPv6 address for the API server, even if I provide a custom IP address in brackets for `kubernetes_url` since it's parsing on colons:

```
[2018/10/25 14:59:26] [ warn] net_tcp_fd_connect: getaddrinfo(host='[fd00'): Temporary failure in name resolution
[2018/10/25 14:59:26] [error] [filter_kube] upstream connection error
```

**Comment by MartinForReal** [comment]
If you enable hostNetwork, It is also important to bind the HOSTNAME env to pod.meta.name as follows:
```
          - name: HOSTNAME
            valueFrom:
              fieldRef:
                apiVersion: v1
                fieldPath: metadata.name
```

**Comment by patrickshan** [comment]
@MartinForReal any reason why we need to set up HOSTNAME environment variable using pod name when using hostNetwork ?

**Comment by MartinForReal** [comment]
please refer to 
https://github.com/fluent/fluent-bit/blob/d97d51371c8da25cb102bdf2f619441a17f4bd26/src/flb_env.c#L50

**Comment by edsiper** [comment]
On IPv6 only environment the option ```ipv6 on``` must be set in the configuration. 

A fix just landed on GIT master and this won't be necessary anymore, the fix will be part of 1.7 release on ~Feb 2021.

ref: c40c1e26eae100084d0747854c1c2a632d07a6cc
